eng/build-utils.ps1

```diff
@@ -200,11 +200,28 @@ function Get-ProjectFile([object]$fileInfo) {
         return $files[0]
       }
       else {
-        $location = Get-Location
-        Set-Location ..
-        if ((Get-Location).Path -eq $location.Path) {
-          # our location didn't change. We must be at the drive root, so give up
-          return $null
+        # We didn't find a project, but lets see if we're in a shared project
+        $files = Get-ChildItem *.shproj
+        if ($files) {
+          # found a shared project, lets see if someone told us what project to use for building etc.
+          $sharedProj = [xml](Get-Content -raw $files[0])
+          $proxyFile = $sharedProj.Project.PropertyGroup | ? { $_.PSobject.Properties.Name -contains "ProxyToConsumingProject" } | Select -Expand ProxyToConsumingProject
+          if ($proxyFile -ne $null) {
+            return Get-Item($proxyFile)
```

Comment:
Don't you need to effectively `Path.Combine` the parent of `$files[0]` and `$proxyFile` here since it's a relative path?

Reply:
I certainly don't claim to understand powershell, but no I don't think so. The `Set-Location` call on line 220 means that this `Get-Item` is done with that parent as the working directory, so as long as the path is relative to the .shrproj file this should work fine.

Reply:
Ah true. This isn't a `-recursive` search and the `Set-Location` will properly root things.